src/context/providers/PlanewaveCutoffsContextProvider.js

```diff
@@ -35,11 +40,52 @@ export class PlanewaveCutoffsContextProvider extends mix(ContextProvider).with(
         return cutoffConfig[this.application.name];
     }
 
+    get _cutoffsFromPseudos() {
+        const pseudos = this.methodData?.pseudo || [];
+        let ecutwfc = 0;
+        let ecutrho = 0;
+
+        pseudos.forEach((data) => {
+            // set the highest cutoff of all elements
+            if (data?.cutoffs?.wavefunction?.standard > ecutwfc) {
+                ecutwfc = data.cutoffs.wavefunction.standard;
+            }
+
+            if (data?.cutoffs?.density?.standard > ecutrho) {
+                ecutrho = data.cutoffs.density.standard;
+            } else {
+                // if rho cutoff is not present, set it based on wfc cutoff
+                // if it is ultrasoft pseudopotential set rho cutoff 8 times
+                // that of wfc cutoff, otherwise 4 times that of wfc cutoff
+                const rhoMultiplier = data?.type === "us" ? 8 : 4;
```

Comment:
Maybe we should isolate this to a separate function, like - `getDensityMultiplierByPseudoType`